engines/wintermute/base/gfx/opengl/base_render_opengl3d.cpp

```diff
@@ -387,58 +421,63 @@ bool BaseRenderOpenGL3D::drawSpriteEx(const OpenGL::Texture &tex, const Wintermu
 		warning("BaseRenderOpenGL3D::SpriteEx y mirroring is not yet implemented");
 	}
 
-	// provide space for 3d position coords, 2d texture coords and a 32 bit color value
-	const int vertexSize = 24;
-	byte vertices[vertexSize * 4] = {};
-
-	float *vertexCoords = reinterpret_cast<float *>(vertices);
+	SpriteVertex vertices[4] = {};
 
 	// texture coords
-	vertexCoords[0 * 6 + 0] = texLeft;
-	vertexCoords[0 * 6 + 1] = texTop;
-	vertexCoords[1 * 6 + 0] = texLeft;
-	vertexCoords[1 * 6 + 1] = texBottom;
-	vertexCoords[2 * 6 + 0] = texRight;
-	vertexCoords[2 * 6 + 1] = texTop;
-	vertexCoords[3 * 6 + 0] = texRight;
-	vertexCoords[3 * 6 + 1] = texBottom;
+	vertices[0].u = texLeft;
+	vertices[0].v = texTop;
+
+	vertices[1].u = texLeft;
+	vertices[1].v = texBottom;
+
+	vertices[2].u = texRight;
+	vertices[2].v = texTop;
+
+	vertices[3].u = texRight;
+	vertices[3].v = texBottom;
 
 	// position coords
-	vertexCoords[0 * 6 + 3] = pos.x - 0.5f;
-	vertexCoords[0 * 6 + 4] = corrected_y - 0.5f;
-	vertexCoords[0 * 6 + 5] = -1.1f;
-	vertexCoords[1 * 6 + 3] = pos.x - 0.5f;
-	vertexCoords[1 * 6 + 4] = corrected_y - height - 0.5f;
-	vertexCoords[1 * 6 + 5] = -1.1f;
-	vertexCoords[2 * 6 + 3] = pos.x + width - 0.5f;
-	vertexCoords[2 * 6 + 4] = corrected_y - 0.5f;
-	vertexCoords[2 * 6 + 5] = -1.1f;
-	vertexCoords[3 * 6 + 3] = pos.x + width - 0.5f;
-	vertexCoords[3 * 6 + 4] = corrected_y - height - 0.5;
-	vertexCoords[3 * 6 + 5] = -1.1f;
+	vertices[0].x = pos.x - 0.5f;
```

Comment:
Might as well un-unroll this loop. Less error-prone that way. The compiler may or may not unroll it as it sees fit.

Reply:
Did this for the color values but I don't see how to put this into a loop without putting width and height into another array and doing some index calculations